projects/solv-protocol-funds/index.js

```diff
@@ -360,6 +361,40 @@ async function ceffuBalance(api, address, graphData) {
   }
 }
 
+async function defiVaultBalance(api, address, graphData) {
+  if (!address[api.chain] || !address[api.chain]["defiVault"]) {
+    return;
+  }
+  let defiVaultData = address[api.chain]["defiVault"];
+
+  let pools = [];
+  for (const graph of graphData.pools) {
+    if (graph['poolId'] == defiVaultData['poolId']) {
+      pools.push(graph)
+    }
+  }
+  if (pools.length > 0) {
+    if (defiVaultData.poolId) {
+      const [totalSupply, poolInfos] = await Promise.all([
+        api.call({ abi: 'erc20:totalSupply', target: defiVaultData.defiVault }),
+        api.call({ abi: abi.poolInfos, target: pools[0].marketContractAddress, params: [defiVaultData.poolId] }),
+      ]);
+
+      const redemptionConcrete = await api.call({
+        abi: abi.concrete,
+        target: poolInfos.poolSFTInfo.openFundRedemption,
+      })
+
+      const [slotTotalValue, nav] = await Promise.all([
+        api.call({ abi: abi.slotTotalValue, target: redemptionConcrete, params: [poolInfos.poolSFTInfo.latestRedeemSlot] }),
+        api.call({ abi: abi.getSubscribeNav, target: poolInfos.navOracle, params: [defiVaultData.poolId, api.timestamp * 1000] }),
```

Comment:
what are these contract methods doing? are the contracts verified? can this be calculated on chain instead? what is counted by defiVaultBalance()?